sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5235234&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5235234&tab=artifacts#/) on release-22.1.0 @ [3abc7eeade10df4b59b0fb24236992f95bf46d9b](https://github.com/cockroachdb/cockroach/commits/3abc7eeade10df4b59b0fb24236992f95bf46d9b):


```
=== RUN   TestExecBuild
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/c0bf42d6e433409ba2f3b1c91d4717d6/logTestExecBuild1936914785
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestExecBuild
    logic.go:4319: -- test log scope end --
--- FAIL: TestExecBuild (125.66s)
=== RUN   TestExecBuild/local/vectorize_local
=== PAUSE TestExecBuild/local/vectorize_local
=== CONT  TestExecBuild/local/vectorize_local
=== CONT  TestExecBuild/local/vectorize_local
    logic.go:3092: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2479/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local:235: SELECT message FROM [SHOW TRACE FOR SESSION]
        WHERE message LIKE 'querying next range at %' OR
              message = '=== SPAN START: kv.DistSender: sending partial batch ==='
        expected:
            querying next range at /Table/109/1/0/0
            querying next range at /Table/109/1/10/0
            querying next range at /Table/109/1/0/0
            === SPAN START: kv.DistSender: sending partial batch ===
            querying next range at /Table/109/1/10/0
            
        but found (query options: "") :
            querying next range at /Table/109/1/0/0
            querying next range at /Table/109/1/0/0
            querying next range at /Table/109/1/0/0
            === SPAN START: kv.DistSender: sending partial batch ===
            querying next range at /Table/109/1/10/0
[09:34:35] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2479/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local with config local: 21 tests, 1 failures
    logic.go:4083: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2479/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local:246: error while processing
    logic.go:4083: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2479/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local:246: too many errors encountered, skipping the rest of the input
[09:34:35] rng seed: 3012838971613641499

[09:34:36] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2479/execroot/cockroach/bazel-out/k8-dbg/bin/pkg/sql/opt/exec/execbuilder/execbuilder_test_/execbuilder_test.runfiles/cockroach/pkg/sql/opt/exec/execbuilder/testdata/hash_sharded_index: 40 statements
        --- FAIL: TestExecBuild/local/vectorize_local (3.12s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81428 sql/opt/exec/execbuilder: TestExecBuild failed [C-test-failure O-robot T-sql-queries branch-master]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-15999

--- a/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local
+++ b/pkg/sql/opt/exec/execbuilder/testdata/vectorize_local
@@ -227,18 +227,21 @@ SET tracing = on; SELECT * FROM tpar WHERE a = 0 OR a = 10
 statement ok
 SET tracing = off
 
-# The span "sending partial batch" means that the scan was parallelized. We're
-# seeing duplicate "querying next range" entries because we first use the range
-# cache to try to partition the spans in order to have parallel TableReaders (we
-# end up with a single partition though), and then we have a single TableReader
-# performing the scan of two spans in parallel.
-query T
-SELECT message FROM [SHOW TRACE FOR SESSION]
+# The span "sending partial batch" means that the scan was parallelized.
+#
+# Most of the time we're seeing duplicate "querying next range" entries because
+# we first use the range cache to try to partition the spans in order to have
+# parallel TableReaders (we end up with a single partition though), and then we
+# have a single TableReader performing the scan of two spans in parallel.
+# However, occasionally the duplicate "querying next range at /Table/109/1/10/0"
+# message is either dropped entirely or replaced with another
+# "querying next range at /Table/109/1/0/0". It's not clear why that happens, so
+# we deduplicate the messages to make the test non-flaky.
+query T rowsort
+SELECT DISTINCT message FROM [SHOW TRACE FOR SESSION]
 WHERE message LIKE 'querying next range at %' OR
       message = '=== SPAN START: kv.DistSender: sending partial batch ==='
 ----
-querying next range at /Table/109/1/0/0
-querying next range at /Table/109/1/10/0
 querying next range at /Table/109/1/0/0
 === SPAN START: kv.DistSender: sending partial batch ===
 querying next range at /Table/109/1/10/0

